### PR TITLE
APPS-3684: DevQA and fix Spryks

### PR DESCRIPTION
**Pull request by Kyrylo-Meretskyi**
[APPS-3684](https://spryker.atlassian.net/browse/APPS-3684)

### CODE REVIEW

File: config/spryk/templates/Zed/Business/DomainEntity/Saver/DomainEntityCreator.php.twig [hunk 1]
```diff
@@ -150,15 +158,15 @@ class {{ domainEntity }}Creator implements {{ domainEntity }}CreatorInterface
...
+            foreach ($valid{{ domainEntity }}CollectionTransfer->get{{ domainEntity | pluralize}}() as ${{ domainEntity | lcfirst }}Transfer) {
+                $this->{{ module | lcfirst }}EntityManager->create{{ domainEntity }}(${{ domainEntity | lcfirst }}Transfer);
             }
 
             // Run post-save plugins
             $valid{{ domainEntity }}CollectionTransfer = $this->execute{{ domainEntity }}PostSavePlugins($valid{{ domainEntity }}CollectionTransfer);
         });
 
-        foreach ($valid{{ domainEntity }}CollectionTransfer->get{{ domainEntity }}s() as ${{ domainEntity | lcfirst }}Transfer) {
+        foreach ($valid{{ domainEntity }}CollectionTransfer->get{{ domainEntity | pluralize}}() as ${{ domainEntity | lcfirst }}Transfer) {
```
Comment (stereomon) [inline config/spryk/templates/Zed/Business/DomainEntity/Saver/DomainEntityCreator.php.twig:169]: Please make sure we always have whitespace before `}}`.

File: config/spryk/spryks/Spryker/Zed/Business/Validation/AddDomainEntityValidator.yml [hunk 2]
```diff
@@ -27,28 +27,15 @@ arguments:
...
     targetPath:
-        value: "src/{{ organization }}/Zed/{{ module }}/Business/Validator/{{ domainEntity }}/"
+        value: "src/{{ organization }}/Zed/{{ module }}/Business/"
 
 preSpryks:
     - AddModule
 
 postSpryks:
     - AddDomainEntityValidatorInterface
-    - AddZedBusinessFactoryMethod:
```
Comment (stereomon) [inline config/spryk/spryks/Spryker/Zed/Business/Validation/AddDomainEntityValidator.yml:39]: Why this get removed?

Comment (sergeyspryker) [reply config/spryk/spryks/Spryker/Zed/Business/Validation/AddDomainEntityValidator.yml:39]: This method is unused. We use specific validators for create and update actions

File: config/spryk/templates/Zed/Persistence/Repository/DomainEntity/GetCollectionMethod.php.twig [hunk 3]
```diff
@@ -1,19 +1,23 @@
...
+
+// Sort
+${{ domainEntity | lcfirst }}Query = $this->applyEntitySorting(${{ domainEntity | lcfirst }}Query, ${{ domainEntity | lcfirst }}CriteriaTransfer);
+
+// Paginate only if the transfer is present
+// This can be different as well, if it not present you might want to fall back to "page 1" by default
+// Again, this is an EXAMPLE, the concrete implementation will depend on each case
+$paginationTransfer = ${{ domainEntity | lcfirst }}CriteriaTransfer->getPagination();
+if ($paginationTransfer !== null) {
+    $objectCollection = $this->applyEntityPagination(${{ domainEntity | lcfirst }}Query, $paginationTransfer);
```
Comment (stereomon) [inline config/spryk/templates/Zed/Persistence/Repository/DomainEntity/GetCollectionMethod.php.twig:15]:
```suggestion
    $objectCollection = $this->apply{{ domainEntity }}Pagination(${{ domainEntity | lcfirst }}Query, $paginationTransfer);
```

File: config/spryk/templates/Zed/Persistence/Repository/DomainEntity/GetCollectionMethod.php.twig [hunk 4]
```diff
@@ -1,19 +1,23 @@
...
-${{ domainEntity | lcfirst }}Entities = $this->apply{{ domainEntity }}Filters(
-    ${{ domainEntity | lcfirst }}Query,
-    ${{ domainEntity | lcfirst }}CriteriaTransfer,
-)->find();
+${{ domainEntity | lcfirst }}Query = $this->getFactory()->create{{ domainEntity }}Query();
 
-if (!${{ domainEntity | lcfirst }}Entities->count()) {
-    return ${{ domainEntity | lcfirst }}CollectionTransfer;
+// Filter
+${{ domainEntity | lcfirst }}Query = $this->applyEntityFilters(${{ domainEntity | lcfirst }}Query, ${{ domainEntity | lcfirst }}CriteriaTransfer);
```
Comment (stereomon) [inline config/spryk/templates/Zed/Persistence/Repository/DomainEntity/GetCollectionMethod.php.twig:5]:
```suggestion
${{ domainEntity | lcfirst }}Query = $this->apply{{ domainEntity }}Filters(${{ domainEntity | lcfirst }}Query, ${{ domainEntity | lcfirst }}CriteriaTransfer);
```

File: config/spryk/templates/Zed/Persistence/Repository/DomainEntity/GetCollectionMethod.php.twig [hunk 5]
```diff
@@ -1,19 +1,23 @@
...
+
+// Sort
+${{ domainEntity | lcfirst }}Query = $this->applyEntitySorting(${{ domainEntity | lcfirst }}Query, ${{ domainEntity | lcfirst }}CriteriaTransfer);
```
Comment (stereomon) [inline config/spryk/templates/Zed/Persistence/Repository/DomainEntity/GetCollectionMethod.php.twig:8]:
```suggestion
${{ domainEntity | lcfirst }}Query = $this->apply{{ domainEntity }}Sorting(${{ domainEntity | lcfirst }}Query, ${{ domainEntity | lcfirst }}CriteriaTransfer);
```

**Review by stereomon — CHANGES_REQUESTED** [review]

File: config/spryk/templates/Zed/Persistence/Repository/DomainEntity/GetCollectionMethod.php.twig (hunk 3 above)
Comment (stereomon) [reply config/spryk/templates/Zed/Persistence/Repository/DomainEntity/GetCollectionMethod.php.twig:15]: Still open

**Review by stereomon — CHANGES_REQUESTED** [review]
We still have some unresolved comments. The rest looks good to me and we only need to wait for the suite-nonsplit review to get the latest feedback.

**Review by stereomon — APPROVED** [review]

**Pull request merged**